yarpmanager: Deprecated warnings

yarpmanager is using several functions deprecated since (I think) Qt 5.9.
We should port it to use non-deprecated API.


```
Scanning dependencies of target yarpmanager
[ 80%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/yarpmanager_autogen/mocs_compilation.cpp.o
[ 80%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/aboutdlg.cpp.o
[ 80%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/applicationviewwidget.cpp.o
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::updateApplicationWindow()’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:638:45: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
  638 |         it->setTextColor(2,QColor("#BF0303"));
      |                                             ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:689:45: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
  689 |         it->setTextColor(2,QColor("#BF0303"));
      |                                             ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:730:49: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
  730 |             it->setTextColor(3,QColor("#BF0303"));
      |                                                 ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:733:49: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
  733 |             it->setTextColor(3,QColor("#BF0303"));
      |                                                 ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘bool ApplicationViewWidget::onRun()’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:1035:53: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 1035 |                 it->setTextColor(2,QColor("#000000"));
      |                                                     ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::runNestedApplication(QTreeWidgetItem*, std::vector<int, std::allocator<int> >*)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:1089:57: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 1089 |                     ch->setTextColor(2,QColor("#000000"));
      |                                                         ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘bool ApplicationViewWidget::onStop()’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:1126:53: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 1126 |                 it->setTextColor(2,QColor("#000000"));
      |                                                     ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::stopNestedApplication(QTreeWidgetItem*, std::vector<int, std::allocator<int> >*)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:1177:57: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 1177 |                     ch->setTextColor(2,QColor("#000000"));
      |                                                         ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘bool ApplicationViewWidget::onKill()’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:1217:53: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 1217 |                 it->setTextColor(2,QColor("#000000"));
      |                                                     ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::killNestedApplication(QTreeWidgetItem*, std::vector<int, std::allocator<int> >*)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:1269:57: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 1269 |                     ch->setTextColor(2,QColor("#000000"));
      |                                                         ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘bool ApplicationViewWidget::onRefresh()’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:1347:53: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 1347 |                 it->setTextColor(2,QColor("#000000"));
      |                                                     ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:1363:49: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 1363 |             it->setTextColor(3,QColor("#000000"));
      |                                                 ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::refreshNestedApplication(QTreeWidgetItem*, std::vector<int, std::allocator<int> >*)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:1389:53: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 1389 |                 ch->setTextColor(2,QColor("#000000"));
      |                                                     ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::updateConnection(int, std::vector<int, std::allocator<int> >&)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:1566:45: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 1566 |         it->setTextColor(2,QColor("#000000"));
      |                                             ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::onSelfConnect(int)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2000:45: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2000 |         it->setTextColor(2,QColor("#008C00"));
      |                                             ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::onSelfDisconnect(int)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2029:45: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2029 |         it->setTextColor(2,QColor("#BF0303"));
      |                                             ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::onSelfResAvailable(int)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2053:49: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2053 |             it->setTextColor(3,QColor("#008C00"));
      |                                                 ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2056:49: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2056 |             it->setTextColor(3,QColor("#008C00"));
      |                                                 ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::onSelfResUnavailable(int)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2082:49: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2082 |             it->setTextColor(3,QColor("#BF0303"));
      |                                                 ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2085:49: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2085 |             it->setTextColor(3,QColor("#BF0303"));
      |                                                 ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::onSelfStart(int)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2097:45: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2097 |         it->setTextColor(2,QColor("#008C00"));
      |                                             ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘void ApplicationViewWidget::onSelfStop(int)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2113:45: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2113 |         it->setTextColor(2,QColor("#BF0303"));
      |                                             ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘virtual void ApplicationViewWidget::onConAvailable(int, int)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2212:84: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2212 |             ui->connectionList->topLevelItem(row)->setTextColor(3,QColor("#008C00"));
      |                                                                                    ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2220:84: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2220 |             ui->connectionList->topLevelItem(row)->setTextColor(4,QColor("#008C00"));
      |                                                                                    ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp: In member function ‘virtual void ApplicationViewWidget::onConUnAvailable(int, int)’:
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2235:84: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2235 |             ui->connectionList->topLevelItem(row)->setTextColor(3,QColor("#BF0303"));
      |                                                                                    ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:2243:84: warning: ‘void QTreeWidgetItem::setTextColor(int, const QColor&)’ is deprecated: Use QTreeWidgetItem::setForeground() instead [-Wdeprecated-declarations]
 2243 |             ui->connectionList->topLevelItem(row)->setTextColor(4,QColor("#BF0303"));
      |                                                                                    ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QTreeWidgetItem:1,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.h:25,
                 from <yarp>/src/yarpmanager/src-manager/applicationviewwidget.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qtreewidget.h:156:17: note: declared here
  156 |     inline void setTextColor(int column, const QColor &color)
      |                 ^~~~~~~~~~~~
[ 80%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/clusterWidget.cpp.o
[ 80%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/customtreewidget.cpp.o
[ 80%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/entitiestreewidget.cpp.o
[ 80%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/genericinfodlg.cpp.o
[ 80%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/genericviewwidget.cpp.o
[ 80%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/logwidget.cpp.o
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/main.cpp.o
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/mainwindow.cpp.o
<yarp>/src/yarpmanager/src-manager/mainwindow.cpp: In member function ‘void MainWindow::onLogError(QString)’:
<yarp>/src/yarpmanager/src-manager/mainwindow.cpp:880:90: warning: ‘virtual void QListWidgetItem::setBackgroundColor(const QColor&)’ is deprecated: Use QListWidgetItem::setBackground() instead [-Wdeprecated-declarations]
  880 |     ui->logWidget->item(ui->logWidget->count() - 1)->setBackgroundColor(QColor("#F9CCCA"));
      |                                                                                          ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QListWidget:1,
                 from <yarp>/src/yarpmanager/src-manager/logwidget.h:22,
                 from /home/ddomenichelli/Workspace/iit/build/yarp/src/yarpmanager/ui_mainwindow.h:30,
                 from <yarp>/src/yarpmanager/src-manager/mainwindow.cpp:20:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qlistwidget.h:124:18: note: declared here
  124 |     virtual void setBackgroundColor(const QColor &color)
      |                  ^~~~~~~~~~~~~~~~~~
<yarp>/src/yarpmanager/src-manager/mainwindow.cpp: In member function ‘void MainWindow::onLogWarning(QString)’:
<yarp>/src/yarpmanager/src-manager/mainwindow.cpp:891:90: warning: ‘virtual void QListWidgetItem::setBackgroundColor(const QColor&)’ is deprecated: Use QListWidgetItem::setBackground() instead [-Wdeprecated-declarations]
  891 |     ui->logWidget->item(ui->logWidget->count() - 1)->setBackgroundColor(QColor("#FFF6C8"));
      |                                                                                          ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QListWidget:1,
                 from <yarp>/src/yarpmanager/src-manager/logwidget.h:22,
                 from /home/ddomenichelli/Workspace/iit/build/yarp/src/yarpmanager/ui_mainwindow.h:30,
                 from <yarp>/src/yarpmanager/src-manager/mainwindow.cpp:20:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qlistwidget.h:124:18: note: declared here
  124 |     virtual void setBackgroundColor(const QColor &color)
      |                  ^~~~~~~~~~~~~~~~~~
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/moduleviewwidget.cpp.o
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/newapplicationwizard.cpp.o
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/resourceviewwidget.cpp.o
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/safe_manager.cpp.o
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/stdoutwindow.cpp.o
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-manager/yscopewindow.cpp.o
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/applicationitem.cpp.o
<yarp>/src/yarpmanager/src-builder/applicationitem.cpp: In constructor ‘ApplicationItem::ApplicationItem(yarp::manager::Application*, yarp::manager::Manager*, QList<int>*, bool, bool, int*, BuilderItem*)’:
<yarp>/src/yarpmanager/src-builder/applicationitem.cpp:50:42: warning: ‘int QFontMetrics::width(const QString&, int) const’ is deprecated: Use QFontMetrics::horizontalAdvance [-Wdeprecated-declarations]
   50 |     textWidth = fontMetric.width(itemName);
      |                                          ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qwidget.h:50,
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qaction.h:46,
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QAction:1,
                 from <yarp>/src/yarpmanager/src-builder/builderitem.h:23,
                 from <yarp>/src/yarpmanager/src-builder/applicationitem.h:22,
                 from <yarp>/src/yarpmanager/src-builder/applicationitem.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qfontmetrics.h:106:9: note: declared here
  106 |     int width(const QString &, int len = -1) const;
      |         ^~~~~
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/arrow.cpp.o
<yarp>/src/yarpmanager/src-builder/arrow.cpp: In member function ‘void Arrow::setConnection(const yarp::manager::Connection&)’:
<yarp>/src/yarpmanager/src-builder/arrow.cpp:133:39: warning: ‘int QFontMetrics::width(const QString&, int) const’ is deprecated: Use QFontMetrics::horizontalAdvance [-Wdeprecated-declarations]
  133 |     textWidth = fontMetric.width(label);
      |                                       ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qwidget.h:50,
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qcombobox.h:44,
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QComboBox:1,
                 from <yarp>/src/yarpmanager/src-builder/arrow.h:69,
                 from <yarp>/src/yarpmanager/src-builder/arrow.cpp:64:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qfontmetrics.h:106:9: note: declared here
  106 |     int width(const QString &, int len = -1) const;
      |         ^~~~~
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/builderitem.cpp.o
[ 81%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/builderscene.cpp.o
[ 82%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/builderwindow.cpp.o
[ 82%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/customtree.cpp.o
[ 82%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/destinationportitem.cpp.o
<yarp>/src/yarpmanager/src-builder/destinationportitem.cpp: In constructor ‘DestinationPortItem::DestinationPortItem(QString, bool, bool, yarp::manager::Application*, BuilderItem*)’:
<yarp>/src/yarpmanager/src-builder/destinationportitem.cpp:42:46: warning: ‘int QFontMetrics::width(const QString&, int) const’ is deprecated: Use QFontMetrics::horizontalAdvance [-Wdeprecated-declarations]
   42 |     int textWidth = fontMetric.width(itemName);
      |                                              ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/qpainter.h:59,
                 from /usr/include/x86_64-linux-gnu/qt5/QtGui/QPainter:1,
                 from <yarp>/src/yarpmanager/src-builder/destinationportitem.h:23,
                 from <yarp>/src/yarpmanager/src-builder/destinationportitem.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qfontmetrics.h:106:9: note: declared here
  106 |     int width(const QString &, int len = -1) const;
      |         ^~~~~
<yarp>/src/yarpmanager/src-builder/destinationportitem.cpp: In member function ‘void DestinationPortItem::editingFinished()’:
<yarp>/src/yarpmanager/src-builder/destinationportitem.cpp:171:46: warning: ‘int QFontMetrics::width(const QString&, int) const’ is deprecated: Use QFontMetrics::horizontalAdvance [-Wdeprecated-declarations]
  171 |     int textWidth = fontMetric.width(itemName);
      |                                              ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/qpainter.h:59,
                 from /usr/include/x86_64-linux-gnu/qt5/QtGui/QPainter:1,
                 from <yarp>/src/yarpmanager/src-builder/destinationportitem.h:23,
                 from <yarp>/src/yarpmanager/src-builder/destinationportitem.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qfontmetrics.h:106:9: note: declared here
  106 |     int width(const QString &, int len = -1) const;
      |         ^~~~~
[ 82%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/itemsignalhandler.cpp.o
[ 82%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/moduleitem.cpp.o
<yarp>/src/yarpmanager/src-builder/moduleitem.cpp: In member function ‘void ModuleItem::init()’:
<yarp>/src/yarpmanager/src-builder/moduleitem.cpp:89:46: warning: ‘int QFontMetrics::width(const QString&, int) const’ is deprecated: Use QFontMetrics::horizontalAdvance [-Wdeprecated-declarations]
   89 |     int textWidth = fontMetric.width(itemName);
      |                                              ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/qpainter.h:59,
                 from /usr/include/x86_64-linux-gnu/qt5/QtGui/QPainter:1,
                 from <yarp>/src/yarpmanager/src-builder/moduleitem.h:24,
                 from <yarp>/src/yarpmanager/src-builder/moduleitem.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qfontmetrics.h:106:9: note: declared here
  106 |     int width(const QString &, int len = -1) const;
      |         ^~~~~
[ 82%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/propertiestable.cpp.o
[ 82%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/sourceportitem.cpp.o
<yarp>/src/yarpmanager/src-builder/sourceportitem.cpp: In constructor ‘SourcePortItem::SourcePortItem(QString, bool, bool, yarp::manager::Application*, BuilderItem*)’:
<yarp>/src/yarpmanager/src-builder/sourceportitem.cpp:43:46: warning: ‘int QFontMetrics::width(const QString&, int) const’ is deprecated: Use QFontMetrics::horizontalAdvance [-Wdeprecated-declarations]
   43 |     int textWidth = fontMetric.width(itemName);
      |                                              ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/qpainter.h:59,
                 from /usr/include/x86_64-linux-gnu/qt5/QtGui/QPainter:1,
                 from <yarp>/src/yarpmanager/src-builder/sourceportitem.h:24,
                 from <yarp>/src/yarpmanager/src-builder/sourceportitem.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qfontmetrics.h:106:9: note: declared here
  106 |     int width(const QString &, int len = -1) const;
      |         ^~~~~
<yarp>/src/yarpmanager/src-builder/sourceportitem.cpp: In member function ‘void SourcePortItem::editingFinished()’:
<yarp>/src/yarpmanager/src-builder/sourceportitem.cpp:175:46: warning: ‘int QFontMetrics::width(const QString&, int) const’ is deprecated: Use QFontMetrics::horizontalAdvance [-Wdeprecated-declarations]
  175 |     int textWidth = fontMetric.width(itemName);
      |                                              ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/qpainter.h:59,
                 from /usr/include/x86_64-linux-gnu/qt5/QtGui/QPainter:1,
                 from <yarp>/src/yarpmanager/src-builder/sourceportitem.h:24,
                 from <yarp>/src/yarpmanager/src-builder/sourceportitem.cpp:19:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qfontmetrics.h:106:9: note: declared here
  106 |     int width(const QString &, int len = -1) const;
      |         ^~~~~
[ 82%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/src-builder/yarpbuilderlib.cpp.o
[ 82%] Building CXX object src/yarpmanager/CMakeFiles/yarpmanager.dir/qrc_res.cpp.o
[ 82%] Linking CXX executable ../../bin/yarpmanager
```

--- a/src/yarplogger/logmodel.cpp
+++ b/src/yarplogger/logmodel.cpp
@@ -372,7 +372,11 @@ void LogModel::addMessages(const std::list<yarp::yarpLogger::MessageEntry> &m_me
     beginInsertRows(QModelIndex(),
                     rowCount(),
                     rowCount() + m_messages.size() - 1);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+    this->m_messages.append(QList<yarp::yarpLogger::MessageEntry>(m_messages.begin(), m_messages.end()));
+#else
     this->m_messages.append(QList<yarp::yarpLogger::MessageEntry>::fromStdList(m_messages));
+#endif
     endInsertRows();
 }
 

--- a/src/yarplogger/messageWidget.cpp
+++ b/src/yarplogger/messageWidget.cpp
@@ -77,14 +77,14 @@ void MessageWidget::addMessage (QString text, int level)
     {
         QString message_to_add = QString ("[WAR] ") + date_s + text;
         addItem(message_to_add);
-        item(this->count() - 1)->setBackgroundColor(QColor("#FFF6C8"));
+        item(this->count() - 1)->setBackground(QColor("#FFF6C8"));
         setCurrentRow(this->count() - 1);
     }
     else if (level == 2)
     {
         QString message_to_add = QString ("[ERR] ") + date_s + text;
         addItem(message_to_add);
-        item(this->count() - 1)->setBackgroundColor(QColor("#F9CCCA"));
+        item(this->count() - 1)->setBackground(QColor("#F9CCCA"));
         setCurrentRow(this->count() - 1);
     }
 }

--- a/src/yarpmanager/src-builder/applicationitem.cpp
+++ b/src/yarpmanager/src-builder/applicationitem.cpp
@@ -47,7 +47,11 @@ ApplicationItem::ApplicationItem(Application* application, Manager *manager,  QL
     customFont.setPointSize(font.pointSize() + 5);
     customFont.setBold(true);
     QFontMetrics fontMetric(customFont);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
+    textWidth = fontMetric.horizontalAdvance(itemName);
+#else
     textWidth = fontMetric.width(itemName);
+#endif
     usedModulesId = usedIds;
 }
 

--- a/src/yarpmanager/src-builder/arrow.cpp
+++ b/src/yarpmanager/src-builder/arrow.cpp
@@ -130,7 +130,11 @@ void Arrow::setConnection(const Connection& conn)
         textLbl.setText(label);
     }
     QFontMetrics fontMetric(font);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
+    textWidth = fontMetric.horizontalAdvance(label);
+#else
     textWidth = fontMetric.width(label);
+#endif
 
     GraphicModel mod = connection.getModelBase();
     if(mod.points.size() > 0){

--- a/src/yarpmanager/src-builder/destinationportitem.cpp
+++ b/src/yarpmanager/src-builder/destinationportitem.cpp
@@ -39,7 +39,11 @@ DestinationPortItem::DestinationPortItem(QString itemName, bool isInApp,
     this->app = app;
 
     QFontMetrics fontMetric(font);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
+    int textWidth = fontMetric.horizontalAdvance(itemName);
+#else
     int textWidth = fontMetric.width(itemName);
+#endif
 
     prepareGeometryChange();
     mainRect = QRectF(-((2*PORT_TEXT_WIDTH) + textWidth)/2,-16,(2*PORT_TEXT_WIDTH) + textWidth,32);
@@ -168,7 +172,11 @@ void DestinationPortItem::editingFinished()
     lineEditWidget->setVisible(false);
 
     QFontMetrics fontMetric(font);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
+    int textWidth = fontMetric.horizontalAdvance(itemName);
+#else
     int textWidth = fontMetric.width(itemName);
+#endif
 
     prepareGeometryChange();
     mainRect = QRectF(-((2*PORT_TEXT_WIDTH) + textWidth)/2,-15,(2*PORT_TEXT_WIDTH) + textWidth,30);

--- a/src/yarpmanager/src-builder/moduleitem.cpp
+++ b/src/yarpmanager/src-builder/moduleitem.cpp
@@ -86,7 +86,11 @@ void ModuleItem::init()
 
 
     QFontMetrics fontMetric(font);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
+    int textWidth = fontMetric.horizontalAdvance(itemName);
+#else
     int textWidth = fontMetric.width(itemName);
+#endif
     int mod = textWidth % 16;
     textWidth+=mod;
 

--- a/src/yarpmanager/src-builder/sourceportitem.cpp
+++ b/src/yarpmanager/src-builder/sourceportitem.cpp
@@ -40,7 +40,11 @@ SourcePortItem::SourcePortItem(QString itemName, bool isInApp,
     this->app = app;
 
     QFontMetrics fontMetric(font);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
+    int textWidth = fontMetric.horizontalAdvance(itemName);
+#else
     int textWidth = fontMetric.width(itemName);
+#endif
 
     prepareGeometryChange();
     mainRect = QRectF(-((2*PORT_TEXT_WIDTH) + textWidth)/2,-16,(2*PORT_TEXT_WIDTH) + textWidth,32);
@@ -172,7 +176,11 @@ void SourcePortItem::editingFinished()
     lineEditWidget->setVisible(false);
 
     QFontMetrics fontMetric(font);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
+    int textWidth = fontMetric.horizontalAdvance(itemName);
+#else
     int textWidth = fontMetric.width(itemName);
+#endif
 
     prepareGeometryChange();
     mainRect = QRectF(-((2*PORT_TEXT_WIDTH) + textWidth)/2,-15,(2*PORT_TEXT_WIDTH) + textWidth,30);

--- a/src/yarpmanager/src-manager/applicationviewwidget.cpp
+++ b/src/yarpmanager/src-manager/applicationviewwidget.cpp
@@ -635,7 +635,7 @@ void ApplicationViewWidget::updateApplicationWindow()
         //it->setFlags(it->flags() | Qt::ItemIsEditable);
         it->setData(0,Qt::UserRole,yarp::manager::MODULE);
         it->setIcon(0,QIcon(":/close.svg"));
-        it->setTextColor(2,QColor("#BF0303"));
+        it->setForeground(2,QColor("#BF0303"));
         ui->moduleList->addTopLevelItem(it);
     }
 
@@ -686,7 +686,7 @@ void ApplicationViewWidget::updateApplicationWindow()
         ui->connectionList->setItemWidget((QTreeWidgetItem *) it,5, comboBox);
         it->setData(0,Qt::UserRole,yarp::manager::INOUTD);
         it->setIcon(0,QIcon(":/disconnect22.svg"));
-        it->setTextColor(2,QColor("#BF0303"));
+        it->setForeground(2,QColor("#BF0303"));
         id++;
     }
 
@@ -727,10 +727,10 @@ void ApplicationViewWidget::updateApplicationWindow()
         it->setData(0,Qt::UserRole,yarp::manager::RESOURCE);
         if (type == "computer") {
             it->setIcon(0,QIcon(":/computer-unavailable22.svg"));
-            it->setTextColor(3,QColor("#BF0303"));
+            it->setForeground(3,QColor("#BF0303"));
         } else {
             it->setIcon(0,QIcon(":/port-unavailable22.svg"));
-            it->setTextColor(3,QColor("#BF0303"));
+            it->setForeground(3,QColor("#BF0303"));
         }
         id++;
     }
@@ -1032,7 +1032,7 @@ bool ApplicationViewWidget::onRun()
 
                 it->setText(2,"waiting");
                 it->setIcon(0,QIcon(":/refresh.svg"));
-                it->setTextColor(2,QColor("#000000"));
+                it->setForeground(2,QColor("#000000"));
 
         }
 
@@ -1054,7 +1054,7 @@ bool ApplicationViewWidget::onRun()
             RIDs.push_back(it->text(1).toInt());
             it->setText(3,"waiting");
             it->setIcon(0,QIcon(":/refresh22.svg"));
-            it->setTextColor(3,QColor("#000000"));
+            it->setForeground(3,QColor("#000000"));
         }
     }*/
 
@@ -1086,7 +1086,7 @@ void ApplicationViewWidget::runNestedApplication(QTreeWidgetItem *it,std::vector
 
                     ch->setText(2,"waiting");
                     ch->setIcon(0,QIcon(":/refresh.svg"));
-                    ch->setTextColor(2,QColor("#000000"));
+                    ch->setForeground(2,QColor("#000000"));
                 }
             }
 
@@ -1123,7 +1123,7 @@ bool ApplicationViewWidget::onStop()
 
                 it->setText(2,"waiting");
                 it->setIcon(0,QIcon(":/refresh.svg"));
-                it->setTextColor(2,QColor("#000000"));
+                it->setForeground(2,QColor("#000000"));
 
             }
         }
@@ -1143,7 +1143,7 @@ bool ApplicationViewWidget::onStop()
             RIDs.push_back(it->text(1).toInt());
             it->setText(3,"waiting");
             it->setIcon(0,QIcon(":/refresh22.svg"));
-            it->setTextColor(3,QColor("#000000"));
+            it->setForeground(3,QColor("#000000"));
         }
     }*/
 
@@ -1174,7 +1174,7 @@ void ApplicationViewWidget::stopNestedApplication(QTreeWidgetItem *it,std::vecto
 
                     ch->setText(2,"waiting");
                     ch->setIcon(0,QIcon(":/refresh.svg"));
-                    ch->setTextColor(2,QColor("#000000"));
+                    ch->setForeground(2,QColor("#000000"));
                 }
             }
 
@@ -1214,7 +1214,7 @@ bool ApplicationViewWidget::onKill()
 
                 it->setText(2,"waiting");
                 it->setIcon(0,QIcon(":/refresh22.svg"));
-                it->setTextColor(2,QColor("#000000"));
+                it->setForeground(2,QColor("#000000"));
 
             }
         }
@@ -1235,7 +1235,7 @@ bool ApplicationViewWidget::onKill()
             RIDs.push_back(it->text(1).toInt());
             it->setText(3,"waiting");
             it->setIcon(0,QIcon(":/refresh22.svg"));
-            it->setTextColor(3,QColor("#000000"));
+            it->setForeground(3,QColor("#000000"));
         }
     }*/
 
@@ -1266,7 +1266,7 @@ void ApplicationViewWidget::killNestedApplication(QTreeWidgetItem *it,std::vecto
 
                     ch->setText(2,"waiting");
                     ch->setIcon(0,QIcon(":/refresh.svg"));
-                    ch->setTextColor(2,QColor("#000000"));
+                    ch->setForeground(2,QColor("#000000"));
                 }
             }
 
@@ -1344,7 +1344,7 @@ bool ApplicationViewWidget::onRefresh()
                                          it->text(7).toLatin1().data());
                 it->setText(2,"waiting");
                 it->setIcon(0,QIcon(":/refresh22.svg"));
-                it->setTextColor(2,QColor("#000000"));
+                it->setForeground(2,QColor("#000000"));
             }
         }
 
@@ -1360,7 +1360,7 @@ bool ApplicationViewWidget::onRefresh()
             resourcesIDs.push_back(it->text(1).toInt());
             it->setText(3,"waiting");
             it->setIcon(0,QIcon(":/refresh22.svg"));
-            it->setTextColor(3,QColor("#000000"));
+            it->setForeground(3,QColor("#000000"));
         }
     }
 
@@ -1386,7 +1386,7 @@ void ApplicationViewWidget::refreshNestedApplication(QTreeWidgetItem *it,std::ve
                 MIDs->push_back(ch->text(1).toInt());
                 ch->setText(2,"waiting");
                 ch->setIcon(0,QIcon(":/refresh22.svg"));
-                ch->setTextColor(2,QColor("#000000"));
+                ch->setForeground(2,QColor("#000000"));
             }
         }
 
@@ -1563,7 +1563,7 @@ void ApplicationViewWidget::updateConnection(int index, std::vector<int>& CIDs)
 
         it->setText(2,"waiting");
         it->setIcon(0,QIcon(":/refresh22.svg"));
-        it->setTextColor(2,QColor("#000000"));
+        it->setForeground(2,QColor("#000000"));
     }
 
 }
@@ -1997,7 +1997,7 @@ void ApplicationViewWidget::onSelfConnect(int which)
     if (it) {
         it->setText(2,"connected");
         it->setIcon(0,QIcon(":/connect22.svg"));
-        it->setTextColor(2,QColor("#008C00"));
+        it->setForeground(2,QColor("#008C00"));
         QString from = it->text(3);
         QString to = it->text(4);
         auto* box = qobject_cast<QComboBox*>(ui->connectionList->itemWidget((QTreeWidgetItem *)it, 5));
@@ -2026,7 +2026,7 @@ void ApplicationViewWidget::onSelfDisconnect(int which)
     if (it) {
         it->setText(2,"disconnected");
         it->setIcon(0,QIcon(":/disconnect22.svg"));
-        it->setTextColor(2,QColor("#BF0303"));
+        it->setForeground(2,QColor("#BF0303"));
         QString from = it->text(3);
         QString to = it->text(4);
         auto* box = qobject_cast<QComboBox*>(ui->connectionList->itemWidget((QTreeWidgetItem *)it, 5));
@@ -2050,10 +2050,10 @@ void ApplicationViewWidget::onSelfResAvailable(int which)
         it->setText(3,"available");
         if (it->text(2) == "computer") {
             it->setIcon(0,QIcon(":/computer-available22.svg"));
-            it->setTextColor(3,QColor("#008C00"));
+            it->setForeground(3,QColor("#008C00"));
         } else {
             it->setIcon(0,QIcon(":/port-available22.svg"));
-            it->setTextColor(3,QColor("#008C00"));
+            it->setForeground(3,QColor("#008C00"));
         }
         /*if (row[m_resColumns.m_col_type] == Glib::ustring("computer"))
             row.set_value(0, m_refPixAvailable);
@@ -2079,10 +2079,10 @@ void ApplicationViewWidget::onSelfResUnavailable(int which)
         it->setText(3,"unavailable");
         if (it->text(2) == "computer") {
             it->setIcon(0,QIcon(":/computer-unavailable22.svg"));
-            it->setTextColor(3,QColor("#BF0303"));
+            it->setForeground(3,QColor("#BF0303"));
         } else {
             it->setIcon(0,QIcon(":/port-unavailable22.svg"));
-            it->setTextColor(3,QColor("#BF0303"));
+            it->setForeground(3,QColor("#BF0303"));
         }
     }
     reportErrors();
@@ -2094,7 +2094,7 @@ void ApplicationViewWidget::onSelfStart(int which)
     if (it) {
         it->setText(2,"running");
         it->setIcon(0,QIcon(":/apply.svg"));
-        it->setTextColor(2,QColor("#008C00"));
+        it->setForeground(2,QColor("#008C00"));
         builder->setModuleRunning(true,which);
         //row[m_modColumns.m_col_editable] = false;
         //row[m_modColumns.m_col_color] = Gdk::Color("#008C00");
@@ -2110,7 +2110,7 @@ void ApplicationViewWidget::onSelfStop(int which)
     if (it) {
         it->setText(2,"stopped");
         it->setIcon(0,QIcon(":/close.svg"));
-        it->setTextColor(2,QColor("#BF0303"));
+        it->setForeground(2,QColor("#BF0303"));
         builder->setModuleRunning(false,which);
     }
     reportErrors();
@@ -2209,7 +2209,7 @@ void ApplicationViewWidget::onConAvailable(int from, int to)
     if (from >= 0) {
         int row;
         if (getConRowByID(from, &row)) {
-            ui->connectionList->topLevelItem(row)->setTextColor(3,QColor("#008C00"));
+            ui->connectionList->topLevelItem(row)->setForeground(3,QColor("#008C00"));
             builder->setOutputPortAvailable(ui->connectionList->topLevelItem(row)->text(3),true);
         }
     }
@@ -2217,7 +2217,7 @@ void ApplicationViewWidget::onConAvailable(int from, int to)
     if (to >= 0) {
         int row;
         if (getConRowByID(to, &row)) {
-            ui->connectionList->topLevelItem(row)->setTextColor(4,QColor("#008C00"));
+            ui->connectionList->topLevelItem(row)->setForeground(4,QColor("#008C00"));
             builder->setInputPortAvailable(ui->connectionList->topLevelItem(row)->text(4),true);
         }
     }
@@ -2232,7 +2232,7 @@ void ApplicationViewWidget::onConUnAvailable(int from, int to)
     if (from >= 0) {
         int row;
         if (getConRowByID(from, &row)) {
-            ui->connectionList->topLevelItem(row)->setTextColor(3,QColor("#BF0303"));
+            ui->connectionList->topLevelItem(row)->setForeground(3,QColor("#BF0303"));
             builder->setOutputPortAvailable(ui->connectionList->topLevelItem(row)->text(3),false);
         }
     }
@@ -2240,7 +2240,7 @@ void ApplicationViewWidget::onConUnAvailable(int from, int to)
     if (to >= 0) {
         int row;
         if (getConRowByID(to, &row)) {
-            ui->connectionList->topLevelItem(row)->setTextColor(4,QColor("#BF0303"));
+            ui->connectionList->topLevelItem(row)->setForeground(4,QColor("#BF0303"));
             builder->setInputPortAvailable(ui->connectionList->topLevelItem(row)->text(4),false);
         }
     }

--- a/src/yarpmanager/src-manager/mainwindow.cpp
+++ b/src/yarpmanager/src-manager/mainwindow.cpp
@@ -877,7 +877,7 @@ void MainWindow::onLogError(QString msg)
 {
     QString text = QString ("[ERR] %1").arg(msg);
     ui->logWidget->addItem(text);
-    ui->logWidget->item(ui->logWidget->count() - 1)->setBackgroundColor(QColor("#F9CCCA"));
+    ui->logWidget->item(ui->logWidget->count() - 1)->setBackground(QColor("#F9CCCA"));
     ui->logWidget->setCurrentRow(ui->logWidget->count() - 1);
 }
 
@@ -888,7 +888,7 @@ void MainWindow::onLogWarning(QString msg)
 {
     QString text = QString ("[WAR] %1").arg(msg);
     ui->logWidget->addItem(text);
-    ui->logWidget->item(ui->logWidget->count() - 1)->setBackgroundColor(QColor("#FFF6C8"));
+    ui->logWidget->item(ui->logWidget->count() - 1)->setBackground(QColor("#FFF6C8"));
     ui->logWidget->setCurrentRow(ui->logWidget->count() - 1);
 }
 

--- a/src/yarpmotorgui/mainwindow.cpp
+++ b/src/yarpmotorgui/mainwindow.cpp
@@ -1340,8 +1340,8 @@ void MainWindow::updateModesTree(PartItem *part)
             jointNode->setText(0,QString("Joint %1").arg(i));
             jointNode->setText(1,mode);
             QColor c = getColorMode(modes.at(i));
-            jointNode->setBackgroundColor(0,c);
-            jointNode->setBackgroundColor(1,c);
+            jointNode->setBackground(0,c);
+            jointNode->setBackground(1,c);
 
 
             if(c == hwFaultColor){
@@ -1378,22 +1378,22 @@ void MainWindow::updateModesTree(PartItem *part)
                 if(item->text(1) != mode){
                     item->setText(1,mode);
                 }
-                if(item->backgroundColor(0) != c){
-                    item->setBackgroundColor(0,c);
-                    item->setBackgroundColor(1,c);
+                if(item->background(0) != c){
+                    item->setBackground(0,c);
+                    item->setBackground(1,c);
                 }
             }
         }
 
         if(!foundFaultPart){
             if(parentNode->data(0,Qt::UserRole).toInt() != TREEMODE_OK){
-                parentNode->setBackgroundColor(0,QColor("white"));
+                parentNode->setBackground(0,QColor("white"));
                 parentNode->setIcon(0,QIcon(":/apply.svg"));
                 parentNode->setData(0,Qt::UserRole,TREEMODE_OK);
             }
         }else{
             if(parentNode->data(0,Qt::UserRole).toInt() != TREEMODE_WARN){
-                parentNode->setBackgroundColor(0,hwFaultColor);
+                parentNode->setBackground(0,hwFaultColor);
                 parentNode->setIcon(0,QIcon(":/warning.svg"));
                 parentNode->setData(0,Qt::UserRole,TREEMODE_WARN);
             }

--- a/src/yarpmotorgui/sequencewindow.cpp
+++ b/src/yarpmotorgui/sequencewindow.cpp
@@ -488,12 +488,12 @@ void SequenceWindow::onStoppedSequence()
 
     if(prevCurrentIndex >=0){
         for(int i=0;i<ui->treePositions->columnCount();i++){
-            ui->treePositions->topLevelItem(prevCurrentIndex)->setBackgroundColor(i,QColor(0,0,0,0));
+            ui->treePositions->topLevelItem(prevCurrentIndex)->setBackground(i,QColor(0,0,0,0));
         }
         ui->treePositions->topLevelItem(prevCurrentIndex)->setIcon(0,QIcon());
 
         for(int i=0;i<ui->treeSpeed->columnCount();i++){
-            ui->treeSpeed->topLevelItem(prevCurrentIndex)->setBackgroundColor(i,QColor(0,0,0,0));
+            ui->treeSpeed->topLevelItem(prevCurrentIndex)->setBackground(i,QColor(0,0,0,0));
         }
         ui->treeSpeed->topLevelItem(prevCurrentIndex)->setIcon(0,QIcon());
 
@@ -506,25 +506,25 @@ void SequenceWindow::onSetCurrentSequenceIndex(int index)
 {
     if(prevCurrentIndex >=0){
         for(int i=0;i<ui->treePositions->columnCount();i++){
-            ui->treePositions->topLevelItem(prevCurrentIndex)->setBackgroundColor(i,QColor(0,0,0,0));
+            ui->treePositions->topLevelItem(prevCurrentIndex)->setBackground(i,QColor(0,0,0,0));
         }
         ui->treePositions->topLevelItem(prevCurrentIndex)->setIcon(0,QIcon());
 
         for(int i=0;i<ui->treeSpeed->columnCount();i++){
-            ui->treeSpeed->topLevelItem(prevCurrentIndex)->setBackgroundColor(i,QColor(0,0,0,0));
+            ui->treeSpeed->topLevelItem(prevCurrentIndex)->setBackground(i,QColor(0,0,0,0));
         }
         ui->treeSpeed->topLevelItem(prevCurrentIndex)->setIcon(0,QIcon());
 
 
     }
     for(int i=0;i<ui->treePositions->columnCount();i++){
-        ui->treePositions->topLevelItem(index)->setBackgroundColor(i,QColor(0,255,0,120));
+        ui->treePositions->topLevelItem(index)->setBackground(i,QColor(0,255,0,120));
 
     }
     ui->treePositions->topLevelItem(index)->setIcon(0,QIcon(":/play.svg"));
 
     for(int i=0;i<ui->treeSpeed->columnCount();i++){
-        ui->treeSpeed->topLevelItem(index)->setBackgroundColor(i,QColor(0,255,0,120));
+        ui->treeSpeed->topLevelItem(index)->setBackground(i,QColor(0,255,0,120));
 
     }
     ui->treeSpeed->topLevelItem(index)->setIcon(0,QIcon(":/play.svg"));

--- a/src/yarpviz/src/portloggerdialog.cpp
+++ b/src/yarpviz/src/portloggerdialog.cpp
@@ -261,7 +261,7 @@ void PortLoggerDialog::startStopLoggers() {
 }
 
 void PortLoggerDialog::MyTimerSlot() {
-    ui->lcdNumberLogTime->display(logTime.elapsed()/1000);
+    ui->lcdNumberLogTime->display(logTime.elapsed()/1000.0);
 }
 
 bool PortLoggerDialog::saveLog(std::string filename, yarp::os::Bottle* samples) {

--- a/src/yarpviz/src/portloggerdialog.h
+++ b/src/yarpviz/src/portloggerdialog.h
@@ -21,7 +21,7 @@
 #include <QDialog>
 #include <yarp/profiler/NetworkProfiler.h>
 #include <QTimer>
-#include <QTime>
+#include <QElapsedTimer>
 
 namespace Ui {
 class PortLoggerDialog;
@@ -50,7 +50,7 @@ private:
     bool isStarted;
     yarp::profiler::graph::Graph *graph;
     QTimer *timer;
-    QTime logTime;
+    QElapsedTimer logTime;
 };
 
 #endif // PORTLOGGERDIALOG_H

